server/aesdsocket.c

```diff
@@ -99,19 +224,142 @@ static void exit_program()
         close(socketfd);
         s_flags.is_socket_open = false;
     }
-    
-    // Close client fd
-    if(s_flags.is_client_fd_open)
-    {
-        close(clientfd);
-        s_flags.is_client_fd_open = false;
-    }  
 
     // Signal caught
     if(s_flags.signal_caught)
     {
         remove(PATH_SOCKETDATA_FILE);
     }
+
+    // Join each thread in the linked-list that hasn't been joined already
+    SLIST_FOREACH_SAFE(t_node, &head, entries, t_node_temp)
+    {
+        if(t_node -> thread_param.thread_complete)
```

Comment:
@ritikar97 on exit you should pthread_join even if the thread is not complete.  Otherwise you won't free resources for a thread which hasn't yet complete before this code is executed.